responsive-spring/src/main/java/dev/responsive/spring/config/ResponsiveFactoryBean.java

```diff
@@ -0,0 +1,353 @@
+import org.springframework.kafka.KafkaException;
+import org.springframework.kafka.config.KafkaStreamsConfiguration;
+import org.springframework.kafka.config.KafkaStreamsCustomizer;
+import org.springframework.kafka.config.KafkaStreamsInfrastructureCustomizer;
+import org.springframework.kafka.config.StreamsBuilderFactoryBean;
+import org.springframework.kafka.core.CleanupConfig;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+public class ResponsiveFactoryBean extends StreamsBuilderFactoryBean {
+
+  public static final Duration DEFAULT_CLOSE_TIMEOUT = Duration.ofSeconds(10);
+
+  private static final LogAccessor LOGGER =
+      new LogAccessor(LogFactory.getLog(StreamsBuilderFactoryBean.class));
+
+  private final ReentrantLock lifecycleLock = new ReentrantLock();
+  private final List<StreamsBuilderFactoryBean.Listener> listeners = new ArrayList<>();
+
+  private KafkaClientSupplier clientSupplier = new DefaultKafkaClientSupplier();
+  private Properties properties;
+  private CleanupConfig cleanupConfig;
+  private KafkaStreamsInfrastructureCustomizer infrastructureCustomizer = new InfraCustomizer();
+
+  private KafkaStreamsCustomizer kafkaStreamsCustomizer;
+  private KafkaStreams.StateListener stateListener;
+  private StateRestoreListener stateRestoreListener;
+  private StreamsUncaughtExceptionHandler streamsUncaughtExceptionHandler;
+  private boolean autoStartup = true;
+  private int phase = Integer.MAX_VALUE - 1000; // NOSONAR magic #
+  private Duration closeTimeout = DEFAULT_CLOSE_TIMEOUT;
+  private boolean leaveGroupOnClose = false;
+  private KafkaStreams kafkaStreams;
+  private volatile boolean running;
+  private Topology topology;
+  private String beanName;
+
+  public ResponsiveFactoryBean() {
+    this.cleanupConfig = new CleanupConfig();
+  }
+
+  public ResponsiveFactoryBean(KafkaStreamsConfiguration streamsConfiguration) {
+    this(streamsConfiguration, new CleanupConfig());
+  }
+
+  public ResponsiveFactoryBean(
+      KafkaStreamsConfiguration streamsConfig,
+      CleanupConfig cleanupConfig
+  ) {
+    super(streamsConfig, cleanupConfig);
+    Assert.notNull(streamsConfig, "'streamsConfig' must not be null");
+    Assert.notNull(cleanupConfig, "'cleanupConfig' must not be null");
+    this.properties = streamsConfig.asProperties();
+    this.cleanupConfig = cleanupConfig;
+  }
+
+  @Override
+  public synchronized void setBeanName(final String name) {
+    this.beanName = name;
+  }
+
+  @Override
+  public void setStreamsConfiguration(Properties streamsConfig) {
+    Assert.notNull(streamsConfig, "'streamsConfig' must not be null");
+    this.properties = streamsConfig;
+  }
+
+  @Override
+  @Nullable
+  public Properties getStreamsConfiguration() {
+    return properties;
+  }
+
+  @Override
+  public void setClientSupplier(final KafkaClientSupplier clientSupplier) {
+    Assert.notNull(clientSupplier, "'clientSupplier' must not be null");
```

Comment:
guess we'll need to remember to update this when the new KafkaClientInterceptor replaces the KafkaClientSupplier -- don't suppose we'll get the deprecation warning to remind us (eg from deprecating the old KafkaStreams constructors which accept the KafkaClientSupplier)?